packages/edge-login-ui-rn/src/components/scenes/ChangePinScene.tsx

```diff
@@ -0,0 +1,183 @@
+      onBack={newAccount ? onBack : undefined}
+      title={newAccount ? s.strings.choose_title_pin : undefined}
+      noUnderline={!newAccount}
+    >
+      <ScrollView ref={scrollViewRef} style={styles.content}>
+        <EdgeText style={styles.description} numberOfLines={2}>
+          {s.strings.pin_desc}
+        </EdgeText>
+        <DigitInput />
+        <View style={styles.actions}>
+          <Fade visible={showNext}>
+            <MainButton
+              label={newAccount ? s.strings.next_label : s.strings.done}
+              type={newAccount ? 'secondary' : 'primary'}
+              onPress={handleNext}
+            />
+          </Fade>
+        </View>
+      </ScrollView>
+    </ThemedScene>
+  )
+}
+
+const getStyles = cacheStyles((theme: Theme) => ({
+  content: {
+    flex: 1,
+    marginHorizontal: theme.rem(0.5),
+    marginTop: theme.rem(1.5)
+  },
+  description: {
+    fontFamily: theme.fontFaceDefault,
+    fontSize: theme.rem(0.875),
+    marginBottom: theme.rem(3.25)
+  },
+  actions: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    marginTop: theme.rem(5),
+    minHeight: theme.rem(3 + 15) // 15 is a hack to avoid the keyboard
+  }
+}))
+
+export const ChangePinScene = connect<StateProps, DispatchProps, OwnProps>(
+  (state: RootState, props: OwnProps) => ({
+    account: props.newAccount ? undefined : getAccount(state),
+    pin: state.create.pin,
+    pinError: state.create.pinError,
+    pinErrorMessage: state.create.pinErrorMessage
+  }),
+  (dispatch: Dispatch, props: OwnProps) => ({
+    onDone() {
+      if (props.newAccount) {
+        dispatch({ type: 'NEW_ACCOUNT_TOS' })
+      } else {
+        Airship.show(bridge => (
+          <ButtonsModal
+            bridge={bridge}
+            title={s.strings.pin_changed}
+            message={s.strings.pin_successfully_changed}
+            buttons={{ ok: { label: s.strings.ok } }}
+          />
+        ))
+          .then(() => dispatch(onComplete()))
+          .catch(showError)
+      }
+    },
+    onBack() {
+      props.newAccount
+        ? dispatch({ type: 'NEW_ACCOUNT_PASSWORD' })
+        : dispatch(onComplete())
+    }
+  })
+)(withTheme(ChangePinSceneComponent))
+
+export const ResecurePinScene = connect<StateProps, DispatchProps, OwnProps>(
+  (state: RootState) => ({
+    account: getAccount(state),
+    pin: state.create.pin,
+    pinError: state.create.pinError,
+    pinErrorMessage: state.create.pinErrorMessage
+  }),
+  (dispatch: Dispatch) => ({
+    onDone() {
+      Airship.show(bridge => (
+        <ButtonsModal
+          bridge={bridge}
+          title={s.strings.pswd_and_pin_changed}
+          message={s.strings.change_pwd_body}
+          buttons={{ ok: { label: s.strings.ok } }}
+        />
+      ))
+        .then(() => dispatch(completeResecure()))
+        .catch(showError)
+    },
+    onBack() {},
+    onSkip() {
+      dispatch(completeResecure())
+    }
+  })
+)(withTheme(ChangePinSceneComponent))
```

Comment:
Since this is functional component, you don't need to use `connect` and `withTheme`.
Instead, use the hooks 'useTheme', 'useSelector', 'useDispatch'.